src/omotes_simulator_core/solver/network/assets/base_item.py

```diff
@@ -35,7 +36,7 @@ def __init__(self, number_of_unknowns: int, name: str, _id: str, number_connecti
         self.number_of_unknowns = number_of_unknowns
         self.number_of_connection_point = number_connection_points
         self.matrix_index = 0
-        self.prev_sol: list[float] = [0.0] * self.number_of_unknowns
+        self.prev_sol = np.zeros(self.number_of_unknowns)
```

Comment:
You need to include prior to the __init__ that the prev_sol is of the type np.ndarray.

`

class BaseItem(ABC):
    """A base class for items in a network."""

   prev_sol: np.ndarray
    """The previous solution of the calculation, defaults to an array of zeros with length (number of unknowns,)

    def __init__(self, number_of_unknowns: int, name: str, _id: str, number_connection_points: int):
`

Reply:
I have included it.